### PR TITLE
fix: some error in restful

### PR DESCRIPTION
**Pull request by smellthemoon**
#37370

### CODE REVIEW

File: internal/distributed/proxy/httpserver/utils.go [hunk 1]
```diff
@@ -801,18 +815,19 @@ func anyToColumns(rows []map[string]interface{}, validDataMap map[string][]bool,
...
 			delete(set, field.Name)
 		}
-
-		if isDynamic {
-			m := make(map[string]interface{})
-			for name, candi := range set {
-				m[name] = candi.v.Interface()
+		// if is not dynamic, but pass more field, will throw err in /internal/distributed/proxy/httpserver/utils.go@checkAndSetData
+		m := make(map[string]interface{})
+		for name, candi := range set {
```
Comment (czs007) [inline internal/distributed/proxy/httpserver/utils.go:820]: if set is empty, do we still need to marshal the set ?

File: internal/distributed/proxy/httpserver/utils.go [hunk 2]
```diff
@@ -801,18 +815,19 @@ func anyToColumns(rows []map[string]interface{}, validDataMap map[string][]bool,
...
+			if name == pkFieldName {
```
Comment (czs007) [inline internal/distributed/proxy/httpserver/utils.go:821]: better to move the check logic to an earlier time.

**Comment by smellthemoon** [comment]
rerun go-sdk

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Pull request merged**